crypto/evp/p_seal.c

```diff
@@ -41,13 +42,20 @@ int EVP_SealInit(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
         goto err;
 
     for (i = 0; i < npubk; i++) {
-        ekl[i] =
-            EVP_PKEY_encrypt_old(ek[i], key, EVP_CIPHER_CTX_key_length(ctx),
-                                 pubk[i]);
-        if (ekl[i] <= 0) {
-            rv = -1;
+        size_t keylen = ekl[i];
```

Comment:
Previously we wrote to ekl unconditionally (without reading it first). This seems to be reading potentially unitialised memory? We also seem to have lost the call to `EVP_CIPHER_CTX_key_length()`

Reply:
Right.  Fixed.